modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/MessagePreProcessor.java

```diff
@@ -130,6 +130,15 @@ private void handleTopicRoutine(InboundEventContainer event, AndesMessage messag
             Set<String> alreadyStoredQueueNames = new HashSet<String>();
             for (AndesSubscription subscription : subscriptionList) {
                 if (!alreadyStoredQueueNames.contains(subscription.getStorageQueueName())) {
+
+                    // Avoid adding QOS 0 MQTT messages to clean session = false subscribers if disconnected
+                    if (AndesSubscription.SubscriptionType.MQTT == subscriptionType
+                            && subscription.isDurable()
```

Comment:
@akalankapagoda, @AnujaLK, @pamod   : I get what you are try to do here. BUT my opinion is to keep the disruptor layer free of protocol specific logic if possible always.
Think about a situation when we bring in all other fancy protocols and adding all of those into kernel it self ( - would be bad right!)?